activerecord/lib/active_record/connection_adapters/abstract/quoting.rb

```diff
@@ -146,7 +146,8 @@ def quote_default_expression(value, column) # :nodoc:
         if value.is_a?(Proc)
           value.call
         else
-          cast_type = column.respond_to?(:cast_type) ? column.cast_type : lookup_cast_type(column.sql_type)
+          # TODO: Remove fetch_cast_type and the need for connection after we release 8.1.
+          cast_type = column.respond_to?(:fetch_cast_type) ? column.fetch_cast_type(self) : lookup_cast_type(column.sql_type)
```

Comment:
This was pre-existing, but I find it weird that we have to use `respond_to?` on a type we're in control of.